src/networking.c

```diff
@@ -0,0 +1,374 @@
+}
+
+
+static gboolean
+cc_oci_netlink_run(const gchar *cmd_line) {
+	gint exit_status = -1;
+	gboolean ret = false;
+
+	ret = g_spawn_command_line_sync(cmd_line,
+				  NULL,
+				  NULL,
+				  &exit_status,
+				  NULL);
+
+	if ( !ret ) {
+		g_critical("failed to spawn [%s]", cmd_line);
+		return false;
+	}
+
+	if (exit_status) {
+		g_critical("netlink failure [%s] [%d]", cmd_line, exit_status);
+		return false;
+	}
+
+	return true;
+}
+
+/*!
+ * Request to create the networking framework
+ * that will be used to connect the specified
+ * container network (veth) to the VM
+ *
+ * The container may be associated with multiple
+ * networks and function has to be invoked for
+ * each of those networks
+ *
+ * Once the OCI spec supports the creation of 
+ * VM compatible tap interfaces in the network
+ * plugin, this setup will not be required
+ *
+ * \param netconfig \c networking configuration
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+cc_oci_network_create (const struct cc_oci_config * const config) {
+	gchar cmd_line[1024];
+
+	/* No networking enabled */
+	if (config->net.ifname == NULL) {
+		return true;
+	}
+
+	if (!cc_oci_tap_create(config->net.tap_device)) {
+		return false;
+	}
+
+	/* Place holder till we create these using appropriate
+	 * netlink commands
+	 */
+
+#define NETLINK_CREATE_BRIDGE "ip link add name %s type bridge"
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_CREATE_BRIDGE,
+		   config->net.bridge);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+#define NETLINK_SET_MAC	"ip link set dev %s address %s"
+	//TODO: Derive non conflicting mac from interface */
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_SET_MAC,
+		   config->net.ifname,
+		   "02:00:CA:FE:00:01");
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+#define NETLINK_ADD_LINK_BR "ip link set dev %s master %s"
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_ADD_LINK_BR,
+		   config->net.ifname,
+		   config->net.bridge);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_ADD_LINK_BR,
+		   config->net.tap_device,
+		   config->net.bridge);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+#define NETLINK_LINK_EN	"ip link set dev %s up"
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_LINK_EN,
+		   config->net.tap_device);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_LINK_EN,
+		   config->net.ifname);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+	g_snprintf(cmd_line, sizeof(cmd_line),
+		   NETLINK_LINK_EN,
+		   config->net.bridge);
+
+	if ( !cc_oci_netlink_run(cmd_line) ) {
+		return false;
+	}
+
+	return true;
+}
+
+static gchar *
+get_address(const gint family, const void * const sin_addr)
+{
+	gchar addrBuf[INET6_ADDRSTRLEN];
+
+	inet_ntop(family, sin_addr, addrBuf, sizeof(addrBuf));
```

Comment:
good catch... will fix that... it should never fail as we pre-validate prior to calling this but we need a check here.